### DISCUSSION TITLE
Create round-trips

### DISCUSSION THREAD

**Discussion by BartFis** [organicmaps/organicmaps#6797] [Ideas]
Would it be possible to create the possibility of automatically creating round-trips based on the distance indicated? This would be ideal for running trips from the house (or new locations) stating 'please present a run for 5K'. There was an app - called eRoutes  - which did this based on OSM data but unfortunately it ceased operations. It would be a great feature exploring new neighbourhoods.

**Comment by Mike-0815** [comment]
Dear developers of OM,

I would love to see the posibility to create fast & simple a ROUND-TRIP from a the actual position with selectable length in km (like TOPO GPS has) !

In the case I don't like the calculated route, I just tap a button and OM will calculate a new one. Distance should be selectable in a wide range. 

That funcionality would be VERY HELPFUL to create within seconds a round-trip. Very useful when you like go for a walk in an area you don't know and find back to your car/shelter etc. ! To make that funcionality PERFECT (even better than Topo GPS), OM could implement the option to include POIs of a selected kind in that round-trip (if it's possible within the selected distance-limit of course, that has priority).

Like this it's done in  "Topo GPS" and this is the only funcion I miss in OM, to forget about "Tops GPS").

Future versions of OM should be able to navigate along:

1) those created routes  
and
2) imported routes (gpx) as well,

but for now it would be very nice, as a first step, to have the posibility to let OM create those round-trips like described before/above.


Kind regards,
Mike

**Comment by fizwidget** [comment]
Agreed this would be cool to have in Organic Maps 👍

In case you’re not aware though, it’s worth having a look at Trail Router in the meantime (this feature is its raison d’être!). It also uses OSM data, and lets you set preferences for things like greenery, hills, and distance 🙂

It even lets you export routes as GPX files, so you could load them into Organic Maps if you wish.

**Reply by Mike-0815** [reply]
Many thanks for the suggestion ! :)

**Reply by Mike-0815** [reply]
Hopefully that funcionality will occur in OM too ...
